[pattgen, dv] Transition to V1 then V2

tunghoang290780 opened the pull request:
Ready for V1 and V2 review

Signed-off-by: Tung Hoang <hoang.tung@wdc.com>

cindychip commented on hw/ip/pattgen/doc/checklist.md:
```diff
@@ -163,24 +163,25 @@ Review        | [V2_CHECKLIST_SCOPED][]               | Done        |
 Regression    | [SIM_NIGHTLY_REGRESSION_V2][]           | Done        |
-Coverage      | [SIM_CODE_COVERAGE_V2][]                | Not Started |
-Coverage      | [SIM_FUNCTIONAL_COVERAGE_V2][]          | Not Started |
+Coverage      | [SIM_CODE_COVERAGE_V2][]                | N/A         |
```
Is there a reason why this is N/A?

tunghoang290780 replied:
Fixed!

weicaiyang reviewed (commented):
we need `DV_DOC_TESTPLAN_REVIEWED` to be done before we sign off V2
Do we need to review meeting for this item? @sriyerg

cindychip commented:
Good point, and the new DV plan needs to have lists of coverpoints.

  Cindy Chen


On Thu, Jun 10, 2021 at 1:52 PM weicaiyang ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> we need DV_DOC_TESTPLAN_REVIEWED to be done before we sign off V2
> Do we need to review meeting for this item? @sriyerg
> <https://github.com/sriyerg>
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/lowRISC/opentitan/pull/6917#pullrequestreview-681248813>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXPOOIYE47JUPEOYSZFQ4LTSEQXHANCNFSM46PCET4Q>
> .
>

weicaiyang commented:
> Good point, and the new DV plan needs to have lists of coverpoints. Cindy Chen

yeah, that's the new requirement. perhaps @tunghoang290780 can use otp_ctrl as example to list all the covergroups in DV plan

tunghoang290780 commented on hw/ip/pattgen/doc/checklist.md:
```diff
@@ -163,24 +163,25 @@ Review        | [V2_CHECKLIST_SCOPED][]               | Done        |
 Integration   | [PRE_VERIFIED_SUB_MODULES_V2][]         | N/A         |
 Issues        | [NO_HIGH_PRIORITY_ISSUES_PENDING][]     | Done        |
 Issues        | [ALL_LOW_PRIORITY_ISSUES_ROOT_CAUSED][] | Done        |
+Integration   | [PRE_VERIFIED_SUB_MODULES_V2][]         | N/A         |
```
Fixed

tunghoang290780 commented:
> > Good point, and the new DV plan needs to have lists of coverpoints. Cindy Chen
> 
> yeah, that's the new requirement. perhaps @tunghoang290780 can use otp_ctrl as example to list all the covergroups in DV plan

@weicaiyang I've updated the DV plan to include covergroups

weicaiyang commented on hw/ip/pattgen/data/pattgen_testplan.hjson:
```diff
@@ -81,4 +81,16 @@
+              - whether channel polarity bit is set/unset
+              - whether channel are run at high/low throughput
+            '''
+    }
```
we can consider to cover every bit of data and predivider are toggled. use auto bins can do it easily. there are some examples [here](https://verificationguide.com/systemverilog/systemverilog-functional-coverage/)

and also cover some value of data length and number of repetitions.

Looks like we should have a review meeting to go over these.

sriyerg replied:
+1 
There are several things about pattgen that can be captured in functional coverage plan. Please work with the designer to come up with a detailed plan. We should have ideally documented it before V1.

sriyerg commented:
The following items are gating the transition to V2:

- [ ] All V1 and V2 tests passing: https://reports.opentitan.org/hw/ip/pattgen/dv/latest/results.html
- [ ] Pattgen coverage not getting generated in the nightly reports - @cindychip 
- [ ] Testplan (and coverage plan review) at V2 (we will deep-dive on coverage configuration & exclusions as well)

sriyerg commented on hw/ip/pattgen/doc/checklist.md:
```diff
@@ -163,18 +163,18 @@ Review        | [V2_CHECKLIST_SCOPED][]               | Done        |
 Testbench     | [ALL_INTERFACES_EXERCISED][]            | Done        |
 Testbench     | [ALL_ASSERTION_CHECKS_ADDED][]          | Done        |
 Testbench     | [SIM_TB_ENV_COMPLETED][]                | Done        |
 Tests         | [SIM_ALL_TESTS_PASSING][]               | Done        |
```
PLease see: https://reports.opentitan.org/hw/ip/pattgen/dv/latest/results.html
There are several failing tests. Please triage and get all V1 and V2 tests passing before marking this as done.

tunghoang290780 replied:
@sriyerg it has been fixed by Cindy. In addition, the alert test was added in PR #7015

sriyerg commented on hw/ip/pattgen/doc/checklist.md:
```diff
@@ -163,18 +163,18 @@ Review        | [V2_CHECKLIST_SCOPED][]               | Done        |
 --------------|-----------------------------------------|-------------|------------------
 Documentation | [DESIGN_DELTAS_CAPTURED_V2][]           | N/A         |
-Documentation | [DV_DOC_COMPLETED][]                    | Not Started |
+Documentation | [DV_DOC_COMPLETED][]                    | Done        |
```
this item is not complete either - the doc still has contains template text.

tunghoang290780 replied:
Fix it

cindychip commented:
Gating item 2). Last Friday we have some hanging and license issue, it should be fixed by today's regression result. 
Weekend regression was killed due to workstation random restart. 
Sorry for the inconvenience!

sriyerg commented on hw/ip/pattgen/data/pattgen.prj.hjson:
```diff
@@ -12,7 +12,7 @@
         life_stage:         "L1",
         design_stage:       "D1",
-        verification_stage: "V1",
+        verification_stage: "V2",
```
Please move this to a separate PR once all checklist items are marked done.

tunghoang290780 replied:
Roll back V1 for review

tunghoang290780 commented:
@sriyerg @weicaiyang @cindychip I addressed some comments and Cindy cleaned the gating items 1,2. Do you think it is enough to have V1 review then layout the missing points for V2

weicaiyang commented:
> @sriyerg @weicaiyang @cindychip I addressed some comments and Cindy cleaned the gating items 1,2. Do you think it is enough to have V1 review then layout the missing points for V2

Looks like we have done more than V1 for this block, we should have a review meeting to make it to V1

tunghoang290780 commented:
> > @sriyerg @weicaiyang @cindychip I addressed some comments and Cindy cleaned the gating items 1,2. Do you think it is enough to have V1 review then layout the missing points for V2
> 
> Looks like we have done more than V1 for this block, we should have a review meeting to make it to V1

@weicaiyang indeed, except more functional coverage plans could be added

tunghoang290780 commented:
@weicaiyang @sriyerg @rasmus-madsen Could this PR enough for a merge then go to V1 review? The other remained issues can be resolved in V2.

weicaiyang commented on hw/ip/pattgen/doc/checklist.md:
```diff
@@ -163,18 +163,18 @@ Review        | [V2_CHECKLIST_SCOPED][]               | Done        |
 Regression    | [SIM_NIGHTLY_REGRESSION_V2][]           | Done        |
-Coverage      | [SIM_CODE_COVERAGE_V2][]                | Not Started |
-Coverage      | [SIM_FUNCTIONAL_COVERAGE_V2][]          | Not Started |
+Coverage      | [SIM_CODE_COVERAGE_V2][]                | Done        |
+Coverage      | [SIM_FUNCTIONAL_COVERAGE_V2][]          | Done        |
```
these 2 items seem not fully done yet.

weicaiyang reviewed (commented):
@tunghoang290780 we can have a testplan review meeting to claim this block to V1.

rasmus-madsen commented:
Closing this PR as Alex is addressing V1 in #7816 
and will create a new PR for the V2 part
